modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/TransactionsInMultipleThreadsTest.java

```diff
@@ -0,0 +1,423 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.distributed;
+
+import java.util.concurrent.Callable;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteTransactions;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.IgniteInterruptedCheckedException;
+import org.apache.ignite.internal.processors.cache.GridCacheAbstractSelfTest;
+import org.apache.ignite.internal.util.lang.GridAbsPredicate;
+import org.apache.ignite.lang.IgniteClosure;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.transactions.Transaction;
+import org.apache.ignite.transactions.TransactionConcurrency;
+import org.apache.ignite.transactions.TransactionIsolation;
+import org.apache.ignite.transactions.TransactionState;
+import org.junit.Assert;
+
+/**
+ *
+ */
+public class TransactionsInMultipleThreadsTest extends AbstractTransactionsInMultipleThreadsTest {
+    /**
+     * @throws Exception  if failed.
+     */
+    public void testSimpleTransactionInAnotherThread() throws IgniteCheckedException {
+        withAllIsolationsAndConcurrencies(new IgniteClosure<Object, Void>() {
+            @Override public Void apply(Object o) {
+                try {
+                    simpleTransactionInAnotherThread();
+                }
+                catch (Exception ignore) {
+                    assert false : "Impossible situation";
```

Comment:
Let's rename "ignore" to "ignired"
Please use fail()
http://junit.sourceforge.net/junit3.8.1/javadoc/junit/framework/Assert.html#fail(java.lang.String)